### PR TITLE
Remove DDG search toggle from private tabs

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/18486

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Clean profile
- Set `brave.use_alternate_private_search_engine` preference to `false`
- Launch browser
- Open Private Tab
- Verify that Private Tab shows new design and no DDG toggle
- Clean profile
- Set `brave.use_alternate_private_search_engine` preference to `true`
- Launch browser
- Open Private Tab
- Verify that Private Tab shows new design and DDG toggle
- Verify that toggle works as expected

### CODE REVIEW

File: browser/ui/webui/new_tab_page/brave_new_tab_message_handler.cc [hunk 1]
```diff
@@ -207,6 +210,9 @@ BraveNewTabMessageHandler* BraveNewTabMessageHandler::Create(
 
 BraveNewTabMessageHandler::BraveNewTabMessageHandler(Profile* profile)
     : profile_(profile), weak_ptr_factory_(this) {
+  if (brave::UseAlternativeSearchEngineProviderEnabled(profile)) {
```
Comment (simonhong) [inline browser/ui/webui/new_tab_page/brave_new_tab_message_handler.cc:213]: We will show new toggle button again even if user turned off by using new toggle button?

Comment (emerick) [reply browser/ui/webui/new_tab_page/brave_new_tab_message_handler.cc:213]: The idea here is that we'll only show the alternative search provider toggle button to the user if they've already enabled it in the past; if they never enabled the alternative search engine (which would be most users), we will no longer show the toggle that enables it on this page.

cc: @rebron to double-check my understanding of how this should work...

**Comment by emerick** [comment]
@petemill Fixed the storybook, sorry about that!

File: browser/search_engines/search_engine_provider_util.h [hunk 2]
```diff
@@ -16,8 +16,12 @@ namespace brave {
 
 bool UseAlternativeSearchEngineProviderEnabled(Profile* profile);
 
+bool ShowAlternativeSearchEngineProviderToggleEnabled(Profile* profile);
+
 void ToggleUseAlternativeSearchEngineProvider(Profile* profile);
 
+void SetShowAlternativeSearchEngineProviderToggle(Profile* profile, bool value);
```
[the remark this answers is not in the document]
Comment (emerick) [reply browser/search_engines/search_engine_provider_util.h]: Good point, fixed.

**Pull request merged**